src/types/utility-types/enumeration.ts

```diff
@@ -31,8 +28,11 @@ export function enumeration<T extends string>(
  * @param options possible values this enumeration can have
  * @returns
  */
-export function enumeration(name: string | string[], options?: any): ISimpleType<string> {
-  const realOptions: string[] = typeof name === "string" ? options! : name
+export function enumeration<T extends string[]>(
+  name: string | T,
+  options?: T
+): ISimpleType<T[number]> {
```

Comment:
Technically not necessary, since the overloads above define the signature, but I find it helps the implementer to have this be the proper signature.